Debug_demo/GameObject.cpp

```diff
@@ -1,26 +1,33 @@
+#include <QVector2D>
 #include "Ball.h"
 #include "GameObject.h"
 
 GameObject::GameObject(const QPointF& pos, const QPointF& speed) :
         pos_(pos), speed_(speed) {}
 
 const QPointF& GameObject::getPos() const {
-    return pos_;
+    return pos_.getValue();
 }
 
 const QPointF& GameObject::getSpeed() const {
-    return speed_;
+    return speed_.getValue();
 }
 
 //FIXME: check const
 void GameObject::calc(int msec) {
-    pos_ += speed_ * (msec / 1000.);
+    speed_.calculate(msec);
+    pos_.calculate(msec);
+    pos_.setValue(pos_.getValue() + speed_.getValue() * (msec / 1000.));
+
+    if (QVector2D(speed_.getValue()).length() < 0.03) {
```

Comment:
This will reset the delta if magnitude of speed vector is less than some threshold. However, the exact threshold value might need some adjustment.